### PR TITLE
Fix/hostname discovery

### PR DESCRIPTION
**Pull request by DimCitus**
See #448.

### CODE REVIEW

File: src/bin/pg_autoctl/ipaddr.c [hunk 1]
```diff
@@ -25,16 +26,22 @@
...
 static bool ipv4eq(struct sockaddr_in *a, struct sockaddr_in *b);
 static bool ipv6eq(struct sockaddr_in6 *a, struct sockaddr_in6 *b);
 static bool fetchIPAddressFromInterfaceList(char *localIpAddress, int size);
+static bool ipaddr_sockaddr_to_string(struct addrinfo *ai,
+									  char *ipaddr, size_t size);
+static bool ipaddr_getsockname(int sock, char *ipaddr, size_t size);
+
 
 /*
  * Connect in UDP to a known DNS server on the external network, and grab our
```
Comment (JelteF) [inline src/bin/pg_autoctl/ipaddr.c:47]: This comment is outdated now. We don't connect to a DNS server on a postgres node and we don't use UDP either.

File: src/bin/pg_autoctl/ipaddr.c [hunk 2]
```diff
@@ -79,31 +128,23 @@ fetchLocalIPAddress(char *localIpAddress, int size,
 		}
 		else
 		{
-			log_warn("Failed to connect: %m");
+			log_warn("Failed to connect to any of the IP addresses for "
+					 "monitor hostname \"%s\" and port %d",
```
Comment (JelteF) [inline src/bin/pg_autoctl/ipaddr.c:132]: 8.8.8.8 is not a monitor
```suggestion
					 "hostname \"%s\" and port %d",
```

Comment (JelteF) [inline src/bin/pg_autoctl/ipaddr.c:132]: Maybe keep the `%m` at the end for debugging?

**Review by JelteF — APPROVED** [review]
Some small comments, but other than that looks good. I pushed one small change to the tmux commands that caused cluster creation to fail on master, but succeed on this branch. This is meant as a regression test of sorts. If you can think of a better way to test this, please do. But this seems better than nothing.

File: src/bin/pg_autoctl/ipaddr.c (hunk 1 above)
Comment (DimCitus) [reply src/bin/pg_autoctl/ipaddr.c]: Good point, fixed now!

File: src/bin/pg_autoctl/ipaddr.c (hunk 2 above)
Comment (DimCitus) [reply src/bin/pg_autoctl/ipaddr.c]: Good point again. Also when given a dotted IP address then `getaddrinfo` does not actually do a DNS query to return several IP addresses, it only does the parsing and prepares the needed binary structure. I have now added a specific message to the case of using `DEFAULT_INTERFACE_LOOKUP_SERVICE_NAME`.

Comment (DimCitus) [reply src/bin/pg_autoctl/ipaddr.c]: It is now done at the connect call site. Also we do `getenv` in between so the errno has changed.

**Pull request merged**